### DISCUSSION TITLE
Constructors on POCO objects, with readonly (NOTNULL) fields.

### DISCUSSION THREAD

**Discussion by j-dc** [linq2db/linq2db#3415] [Q&A] [answered]
When a field is NotNull I would like to see that reflected in my POCO class as a 'read only' property.
Especially considering the new 'null state analysis':  https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references

Can I do this with Linq2db ?

``` c#
    [Table()]
    public partial class MyTable {

        public MyTable(string id) {
            ID = id;
        }


        [PrimaryKey, NotNull] public string ID { get; }
        [Column()] public string? Name { get; set; }
    }

```

**Comment by MaceWindu** [comment, marked as answer]
Don't think it's documented, but from code I see that when there is no default constructor defined, we choose constructor with max number of parameters with names matching column name (case-insensitive comparison by default, but could be changed using `dataConnection.MappingSchema.ColumnNameComparer` property). Note that column name used in comparison, not mapped member name (probably some really old code...)

Maybe @sdanyliv could add more details here, as I wasn't aware if this feature and currently just looking at existing code. Could miss some details.
